builder/image_solver.go

```diff
@@ -297,14 +293,13 @@ func (m *multiImageSolver) addRefToResult(ctx context.Context, gwClient gwclient
 		return errors.Wrap(err, "initial state to ref conversion")
 	}
 
-	refKey := fmt.Sprintf("image-%s", imageDef.ImageName)
-	refPrefix := fmt.Sprintf("ref/%s", refKey)
-	res.AddRef(refKey, ref)
+	refPrefix, err := gwCrafter.AddPushImageEntry(ref, imageIndex, imageDef.ImageName, false, false, saveImage.Image, nil)
+	if err != nil {
+		return err
+	}
 
 	localRegPullID := fmt.Sprintf("sess-%s/%s", gwClient.BuildOpts().SessionID, imageDef.ImageName)
-	res.AddMeta(fmt.Sprintf("%s/export-image-local-registry", refPrefix), []byte(localRegPullID))
-	res.AddMeta(fmt.Sprintf("%s/%s", refPrefix, exptypes.ExporterImageConfigKey), imgJSON)
-	res.AddMeta(fmt.Sprintf("%s/image.name", refPrefix), []byte(imageDef.ImageName))
+	gwCrafter.AddMeta(fmt.Sprintf("%s/export-image-local-registry", refPrefix), []byte(localRegPullID))
```

Comment:
This too will eventually be moved into the gatewaycrafter, but in a different PR.